misk-crypto/src/main/kotlin/misk/crypto/Cipher.kt

```diff
@@ -0,0 +1,68 @@
+package misk.crypto
+
+import com.google.crypto.tink.Aead
+import com.google.crypto.tink.JsonKeysetWriter
+import com.google.crypto.tink.KeysetHandle
+import com.google.crypto.tink.aead.AeadFactory
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import java.io.ByteArrayOutputStream
+import java.util.Base64
+
+/**
+ * An even simpler interface than Tink's [Aead].
+ * This also makes it easier to mock Ciphers for testing purposes.
+ */
+interface Cipher {
+  /**
+   * Encrypt the given data.
+   */
+  fun encrypt(plaintext: ByteString): ByteString
```

Comment:
The methods in this interface omit the associated data arguments to encrypt and decrypt supported by Tink's Aead interface. The associated data used by AEADs can be used to add integrity validation of non-secret data in a single pass for higher performance and less moving parts than having to hash them separately. If there isn't any associated data, they can just be null for both encrypt and decrypt. What is the trade-off here for not surfacing those capabilities from misk-crypto?

Reply:
Also: does using `ByteString` objects here cause additional copies of the plaintext to be created on the heap? Normally, crypto libraries try to be careful to not make additional copies of plaintext data so that the caller can overwrite the plaintext after calling `encrypt` and are done with it.

Reply:
Sorry I didn't address this earlier.

I think that for now, having the simple encrypt/decrypt interface with just the data should be enough, but I'm definitely open to expanding this API to do validation as well.
I didn't add it here for now, because I want to keep this initial PR simple and with few features.
We can add this and also the HMAC validation as well, after we'll get a little bit of milage on this module.

As for the `ByteString` objects, I'm not really sure, but I bet @swankjesse knows about it.